pkg/contracts/src/CVStrategyV0_0.sol

```diff
@@ -0,0 +1,976 @@
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import {console} from "forge-std/console.sol";
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+import {UUPSUpgradeable} from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import {OwnableUpgradeable} from "openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import {BaseStrategyUpgradeable} from "./BaseStrategyUpgradeable.sol";
+
+interface IPointStrategy {
+    function deactivatePoints(address _member) external;
+    function increasePower(address _member, uint256 _amountToStake) external returns (uint256);
+    function decreasePower(address _member, uint256 _amountToUntake) external returns (uint256);
+    function getPointSystem() external returns (StrategyStruct.PointSystem);
+}
+
+library StrategyStruct {
+    enum ProposalType {
+        Signaling,
+        Funding,
+        Streaming
+    }
+
+    enum PointSystem {
+        Fixed,
+        Capped,
+        Unlimited,
+        Quadratic
+    }
+
+    struct CreateProposal {
+        // uint256 proposalId;
+        uint256 poolId;
+        address beneficiary;
+        // ProposalType proposalType;
+        uint256 amountRequested;
+        address requestedToken;
+        Metadata metadata;
+    }
+
+    enum ProposalStatus {
+        Inactive, // Inactive
+        Active, // A vote that has been reported to Agreements
+        Paused, // A vote that is being challenged by Agreements
+        Cancelled, // A vote that has been cancelled
+        Executed // A vote that has been executed
+
+    }
+
+    struct Proposal {
+        uint256 proposalId;
+        uint256 requestedAmount;
+        uint256 stakedAmount;
+        uint256 convictionLast;
+        address beneficiary;
+        address submitter;
+        address requestedToken;
+        uint256 blockLast;
+        ProposalStatus proposalStatus;
+        mapping(address => uint256) voterStakedPoints; // voter staked points
+        Metadata metadata;
+    }
+
+    struct ProposalSupport {
+        uint256 proposalId;
+        int256 deltaSupport; // use int256 to allow negative values
+    }
+
+    struct PointSystemConfig {
+        //Capped point system
+        uint256 maxAmount;
+    }
+
+    struct SybilConfig {
+        address sybilOracleScorer;
+        uint256 threshold;
+    }
+
+    struct InitializeParams {
+        address registryCommunity;
+        // Alpha | Decay | a
+        uint256 decay;
+        // MaxRatio | Beta | b | SpendingLimit
+        uint256 maxRatio;
+        // Weight | RHO | p
+        uint256 weight;
+        uint256 minThresholdPoints;
+        // Proposal Type
+        ProposalType proposalType;
+        //NEXT: use this for tests
+        PointSystem pointSystem;
+        PointSystemConfig pointConfig;
+        SybilConfig sybilConfig;
+    }
+}
+
+contract CVStrategyV0_0 is OwnableUpgradeable, BaseStrategyUpgradeable, IPointStrategy, ERC165 {
```

Comment:
So this will be bumped each times we change the implementation ?

Reply:
Yeah V0_0 will be V0_1 if that dont need others dependencies also need be bumped, if that the case so that should be V1_0 instead. 

I have a draw in the excalidraw explaing it